zircon.core/src/commonMain/kotlin/org/hexworks/zircon/internal/fragment/impl/table/TableColumn.kt

```diff
@@ -0,0 +1,42 @@
+package org.hexworks.zircon.internal.fragment.impl.table
+
+import org.hexworks.zircon.api.Components
+import org.hexworks.zircon.api.component.Component
+
+/**
+ * This class represents the definition of a column in a table. It provides means to get the value
+ * from a model object and wrap that value into a [Component]. This component is then displayed
+ * as table cell.
+ *
+ * @param M the type of the model. In other words: Every element/row in the table has this type
+ * @param V the type of the value of each cell in this column
+ * @param C type of the [Component] used to represent each cell
+ */
+open class TableColumn<M : Any, V : Any, C : Component>(
+    /**
+     * The name of this column. Will be used as table header.
+     */
+    val name: String,
+    /**
+     * The width of this column. The component created by [componentCreator] may not be wider than this.
+     */
+    val width: Int,
+    private val valueAccessor: (M) -> V,
+    private val componentCreator: (V) -> C
```

Comment:
If I'm reading the first modified file correctly, we do something special if `V` is an ObservableValue. Should we document that here somewhere?

Reply:
It is only treated specially in one implementation: `Columns.textColumn()` and it [is documented there](https://github.com/Hexworks/zircon/pull/368/files#diff-1406f6b4076df7701ddab7e04431ecd2ae3594392f489092320aefe12db4d7a4R21) ;)